.github/workflows/on-push-do-docs.yml

```diff
@@ -0,0 +1,45 @@
+name: on-push-do-docs
+
+on:
+  push:
+    branches: [main]
+    paths: [ "samples/Snippets/**" ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  update-docs:
+    name: update-docs
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+
+      - name: Setup .NET SDK
+        uses: actions/setup-dotnet@3447fd6a9f9e57506b15f895c5b76d3b197dc7c2 # v3.2.0
+
+      - name: Run MarkdownSnippets
+        shell: bash
+        env:
+          DOTNET_CLI_TELEMETRY_OPTOUT: true
+          DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+        run: |
+          dotnet tool restore
+          dotnet mdsnippets
+
+      - name: Push changes
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        shell: bash
+        run: |
+          git config --local user.email "138034000+polly-updater-bot[bot]@users.noreply.github.com"
+          git config --local user.name "polly-updater-bot[bot]"
+          git commit -m "Update documentation\n\nUpdate examples in documentation." -a  || echo "Nothing to commit"
+          remote="https://${GITHUB_ACTOR}:${{ secrets.GITHUB_TOKEN }}@github.com/${GITHUB_REPOSITORY}.git"
+          git push "${remote}" "${GITHUB_REF_NAME}" || echo "Nothing to push"
```

Comment:
I'm not sure this will work due to required statuses and protected branches.

I think we'll need to have it open a pull request from a branch.

Reply:
That's true, any tips on how to do this?

Reply:
Here's a copy-pasta snippet I've adjusted where relevant from an internal repo I have that periodically checks an external resource for changes to a JSON file in a repo and opens a PR if there are changes:

```yaml
    - name: Update documentation
      id: update-docs
      shell: pwsh
      run: |
        $ErrorActionPreference = "Stop"
        $ProgressPreference = "SilentlyContinue"

        // Stuff that regenerates the content happens here

        $GitStatus = (git status --porcelain)
        if ([string]::IsNullOrEmpty($GitStatus)) {
          Write-Host "No changes to commit."
          exit 0
        }

        $BranchName = "my-branch-name"
        $GitEmail = "138034000+polly-updater-bot[bot]@users.noreply.github.com"
        $GitUser = "polly-updater-bot[bot]"

        git config user.email $GitEmail | Out-Null
        git config user.name $GitUser | Out-Null
        git remote set-url "${{ github.server_url }}/${{ github.repository }}.git" | Out-Null
        git fetch origin | Out-Null
        git rev-parse --verify --quiet ("remotes/origin/" + $BranchName) | Out-Null

        if ($LASTEXITCODE -eq 0) {
          Write-Host "Branch $BranchName already exists."
          exit 0
        }

        git checkout -b $BranchName
        git add .
        git commit -m "COMMIT MESSAGE GOES HERE"
        git push -u origin $BranchName
        "updated-docs=true" >> $env:GITHUB_OUTPUT

    - name: Generate GitHub application token
      if: steps.update-docs.outputs.updated-docs == 'true'
      id: generate-application-token
      uses: peter-murray/workflow-application-token-action@8e1ba3bf1619726336414f1014e37f17fbadf1db # v2.1.0
      with:
        application-id: ${{ secrets.POLLY_UPDATER_BOT_APP_ID }}
        application-private-key: ${{ secrets.POLLY_UPDATER_BOT_KEY }}
        permissions: "contents:write, pull_requests:write"

    - name: Create pull request
      if: steps.update-docs.outputs.updated-docs == 'true'
      uses: actions/github-script@d7906e4ad0b1822421a7e6a35d5ca353c962f410 # v6.4.1
      with:
        github-token: ${{ steps.generate-application-token.outputs.token }}
        script: |
          const { repo, owner } = context.repo;
          const workflowUrl = `${{ github.server_url }}/${owner}/${repo}/actions/runs/${process.env.GITHUB_RUN_ID}`;
          const result = await github.rest.pulls.create({
            title: 'PR TITLE HERE',
            owner,
            repo,
            head: 'BRANCH NAME HERE',
            base: 'main',
            body: [
              'PR DESCRIPTION HERE',
              '',
              `This pull request was generated by [GitHub Actions](${workflowUrl}).`
            ].join('\n')
          });
```

Reply:
Thanks. I added these changes and it creates a PR:

[git status --porcelain](https://github.com/App-vNext/Polly/pull/1545)